File: 06-Pila/java/src/StackNode.java
public class StackNode<T> {
    private Node<T> top;
    private int length;

    public StackNode() {
        top = null;
        length = 0;
    }

    public boolean isEmpty() {
        return top == null && length == 0;
    }

    public int length() {
        return length;
    }

    public T pop() {
        if(isEmpty()) {
            return null;
        }

        T topValue = top.getData();
        top = top.getNext();
        length--;
        return topValue;
    }

    public T peek() {
        if(isEmpty()) {
            return null;
        }

        return top.getData();
    }

    public void push(T value) {
        Node<T> newNode = new Node<T>(value);
        newNode.setNext(top);
        top = newNode;
        length++;
    }

    @Override
    public String toString() {
        String stackString = "";
        Node<T> iteratorNode = top;
        while(iteratorNode != null){
            stackString += iteratorNode;
            iteratorNode = iteratorNode.getNext();
        }
        return stackString;
    }
}

File: 06-Pila/python/stackNode.py
from node import Node

class StackNode:
    top = None
    length = 0;

    def __init__(self):
        self.length = 0
        self.top = None

    def isEmpty(self):
        return self.top == None and self.length == 0

    def getLength(self):
        return self.length

    def push(self, valor):
        newNode = Node(valor, self.top)
        self.top = newNode
        self.length += 1

    def pop(self):
        if self.isEmpty():
            return
        temp = self.top
        self.top = self.top.getNext()
        self.length -= 1
        return temp.getData()

    def peek(self):
        if self.isEmpty():
            return
        return self.top.getData()

    def printStack(self):
        current = self.top
        while current != None:
            print(current.getData())
            current = current.getNext()

    def __str__(self):
        current = self.top
        cadena = ""
        while current != None:
            cadena += str(current)
            current = current.getNext()
        return cadena
